internal/tasks/process-file-updates-task/process-file-updates-task.yaml

```diff
@@ -243,7 +243,61 @@ spec:
         fi
 
         echo -e "\n*** START LOCAL CHANGES ***\n"
-        git diff
+        git diff | tee "${TEMP}"/tempMRFile.diff
+
+        if [[ -s "${TEMP}"/tempMRFile.diff ]]; then
+            # replacements exist
+            awk '/^@@/ {match($0, /@@ ([^@]+) @@/, arr); print arr[1]}' "${TEMP}"/tempMRFile.diff |\
+               tee "${TEMP}"/lineFile
+        fi
+
+        openMRList=$(glab mr list -R "${UPSTREAM_REPO}" --search "Konflux release" |grep "^!"| cut -f1 | tr -d '!')
+        for mrNum in ${openMRList} ; do
+            foundMR=false
+            glab mr diff "${mrNum}" --repo "${UPSTREAM_REPO}" > "${TEMP}"/oneMR.diff 
+
+            if [[ ! -s "${TEMP}"/lineFile ]]; then
+                # only seed files  
+                foundMR=false
+                grep "^diff --git" "${TEMP}"/oneMR.diff | tee "${TEMP}"/tmpFile
+                while read -r line ; do
+                    changedFileName=$(echo "$line" | awk '{sub(/^b\//, "", $NF); print $NF}')
+                    if [[ "${changedFileName}" != "${targetFile}" ]]; then
+                        continue
+                    fi
+                    foundMR=true
+                    break
+                done < "${TEMP}"/tmpFile
+
+                if [[ "$foundMR" == false ]]; then
+                    continue
+                fi
+
+                grep '^[+][^+]' "${TEMP}"/oneMR.diff | sed -E 's/^[+]//' | tee "${TEMP}/mrFile"
+                if  diff -q "${TEMP}"/mrFile "${targetFile}" > /dev/null; then
+                    foundMR=true
+                else
+                    foundMR=false
+                fi
+            else
+                # replacements exist
+                grep '^[-+@]' "${TEMP}"/oneMR.diff | sed -E 's/^[@]//; s/@@.*//' | tee "${TEMP}/mrFile"
+                grep '^[-+@]' "${TEMP}"/tempMRFile.diff | sed -E 's/^[@]//; s/@@.*//' | tee "${TEMP}/tmpFile"
+                if diff -q "${TEMP}"/mrFile "${TEMP}"/tmpFile > /dev/null; then
+                    foundMR=true
+                else
+                    foundMR=false
+                fi
+            fi
+            # if all the lines are matched, the MR is the same one
+            # it should exit 0 if the same MR exists 
+            if [[ "$foundMR" == true ]]; then
```

Comment:
Could you add another echo here saying that an existent MR for the same change was found, so the pipeline user would know why his new MR was not created.

Reply:
Yes. Added.